webapp/lib/view.dart

```diff
@@ -1019,3 +1024,41 @@ class UrlView {
     return false;
   }
 }
+
+enum SnackbarNotificationType {
+  info,
+  success,
+  warning,
+  error
+}
+
+class SnackbarView {
+  DivElement snackbarElement;
+  DivElement _contents;
+
+  SnackbarView() {
+    snackbarElement = new DivElement()
+      ..id = 'snackbar'
+      ..classes.add('hidden')
+      ..title = 'Click to close notification.'
+      ..onClick.listen((_) => hideSnackbar());
+
+    _contents = new DivElement()
+      ..classes.add('contents');
+    snackbarElement.append(_contents);
+  }
+
+  showSnackbar(String message, SnackbarNotificationType type) {
+    _contents.text = message;
+    snackbarElement.classes.remove('hidden');
+    snackbarElement.setAttribute('type', type.toString().replaceAll('SnackbarNotificationType.', ''));
+    new Timer(new Duration(seconds: 3), () => hideSnackbar());
```

Comment:
Could you a couple of these magic numbers into named constants defined on the class? E.g. I think this is the amount of time that the snackbar will be visible for?

Reply:
Good idea, done now!